packages/nx/src/command-line/release/publish.ts

```diff
@@ -198,3 +199,22 @@ async function runPublishOnProjects(
     }
   }
 }
+
+function ensureAllProjectsHaveTarget(projectsToRun: ProjectGraphProjectNode[]) {
+  const requiredTargetName = 'nx-release-publish';
+  const projectsMissingTarget = projectsToRun.filter(
+    (project) => !projectHasTarget(project, requiredTargetName)
+  );
+  if (projectsMissingTarget.length === 0) {
+    return;
+  }
+  output.error({
+    title: `Based on your config, the following projects were matched for publishing but do not have the "${requiredTargetName}" target specified:`,
+    bodyLines: [
+      ...projectsMissingTarget.map((p) => `- ${p.name}`),
+      '',
+      'There are a few possible reasons for this: (1) The projects may be private (2) You may not have an appropriate plugin (such as `@nx/js`) installed which adds the target automatically to public projects (3) You intended to configure the target manually, or exclude those projects via config in nx.json',
+    ],
+  });
+  process.exit(1);
```

Comment:
Could this have been thrown as an error? It's quite a lengthy message but since we have a programmatic API, it's weird for their process to suddenly exit after calling the function without a chance to `catch` it?

Reply:
Yes fair point, it would block programmatic consumers. We don’t have handleErrors in use in this command so I’ll double check the experience tomorrow

Reply:
So the way I have decided to handle this is leverage the handleErrors function for the CLI usage like we do for the other subcommands, but additionally have an appreciation for whether or not the code is being run via the CLI.

This allows us to keep the CLI error messaging focused on the publish run (where there are no other config errors), instead of having the runCommand failure and then another error message afterwards just reiterating that publishing has failed.

E.g. what I have gone for 

![image](https://github.com/nrwl/nx/assets/900523/cd7f7cb1-799c-4639-a4f9-abd8ef224cbd)

vs

no appreciation for how it's being run via the CLI

![image](https://github.com/nrwl/nx/assets/900523/69a7af3f-9b24-4066-92e4-81abf91cd69c)